python/tank/util/metrics.py

```diff
@@ -282,6 +282,7 @@ def _dispatch(self, metrics):
             },
             "metrics": [m.data for m in metrics]
         }
+        self._engine.log_debug("Logging metrics: %s" % ([m.data for m in metrics],))
```

Comment:
This is running in a bg thread right? Do you think it could be causing instability in some engines? (Maybe we shouldn't be concerned about this here, but instead fix those engines). In the past, we have adopted a philosophy of trying to do as little as possible in threads, and stay as quite as possible during those phases, but that stems from an era of both toolkit and DCCs where things (DCCs, Pyside, everything) was more brittle, so not sure it's relevant any longer. What do you think?